src/App.js

```diff
@@ -0,0 +1,216 @@
+import Landing from './components/Landing'
+import Catalog from './components/Catalog'
+import MovieDetails from './components/MovieDetails'
+import Nav from './components/Nav'
+import moviesData from './data/movies'
+import './App.css'
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      users: [
+        {
+          id: 0,
+          name: 'default',
+          color: 'gray',
+          rentedMovies: [],
+          budget: 10,
+        },
+        ...this.getUsersFromStorage(),
+      ],
+      budget: 10,
+      movies: moviesData,
+    } // endof State
+  }
+  deleteUser = userId => {
+    userId = parseInt(userId)
+    console.log('userId', userId)
+    const currentUsers = [...this.state.users]
+    console.log('currentUsers', currentUsers)
+    const filteredUsers = currentUsers.filter(
+      user => parseInt(user.id) !== parseInt(userId)
+    )
+    console.log('filteredUsers', filteredUsers)
+    this.setState(
+      {
+        users: filteredUsers,
+      },
+      () => {
+        this.removeUserFromStorage(userId)
+      }
+    )
+  }
+  removeUserFromStorage = userId => {
+    localStorage.setItem(
+      'reflixStorage',
+      JSON.stringify({
+        loggedUser: this.state.loggedUser,
+        users: this.state.users.filter(user => user.id !== 0),
+      })
+    )
+  }
+  saveNewUser = (username, color) => {
+    let maxId = 0
+    this.state.users.forEach(user => {
+      if (user.id > maxId) maxId = user.id + 1
+    })
+    ++maxId
+
+    const newUser = {
+      id: maxId,
+      name: username,
+      color: color,
+      rentedMovies: [],
+      budget: 10,
+    }
+    const currentUsers = [...this.state.users]
+    currentUsers.push(newUser)
+    this.setState(
+      {
+        users: currentUsers,
+      },
+      () => {
+        this.insertUserToStorage(newUser.id)
+      }
+    )
+  }
+  getUsersFromStorage = () => {
+    if (!localStorage.getItem('reflixStorage')) {
+      localStorage.setItem(
+        'reflixStorage',
+        JSON.stringify({ loggedUser: this.state.loggedUser, users: [] })
+      )
+    }
+    const reflixStorage = JSON.parse(localStorage.getItem('reflixStorage'))
+
+    return reflixStorage.users
+  }
+  insertUserToStorage = userId => {
+    if (userId !== 0) {
+      const userDetails = this.state.users.find(user => user.id === userId)
+      const storageUsers = this.getUsersFromStorage()
+      const usersWithout = storageUsers.filter(
+        user => user.id !== userDetails.id
+      )
+      usersWithout.push(userDetails)
+      localStorage.setItem(
+        'reflixStorage',
+        JSON.stringify({
+          loggedUser: this.state.loggedUser,
+          users: usersWithout,
+        })
+      )
+    }
+  }
+  logUserIn = userId => {
+    if (!localStorage.getItem('reflixStorage')) {
+      localStorage.setItem(
+        'reflixStorage',
+        JSON.stringify({ loggedUser: this.state.loggedUser, users: [] })
+      )
+    }
```

Comment:
this little piece of code (lines 109-114) also appear in lines 81-86 => you could make it a small method of something like "handleEmptyStorage" or something like that)